File: app/src/main/java/com/github/willjgriff/ethereumwallet/di/modules/EthereumNodeModule.java
package com.github.willjgriff.ethereumwallet.di.modules;

import android.content.Context;

import com.github.wiljgriff.ethereumwallet.ethereum.node.EthereumNode;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Will on 16/03/2017.
 */

@Module
public class EthereumNodeModule {

	@Provides
	@Singleton
	@Named("ethereum_path")
	String provideEthereumPath(Context context) {
		return context.getFilesDir().toString() + "/ethereum/";
	}

	@Provides
	@Singleton
	EthereumNode provideEthereum(@Named("ethereum_path") String ethereumPath) {
		return new EthereumNode(ethereumPath);
	}
}

File: app/src/main/kotlin/com/github/wiljgriff/ethereumwallet/di/modules/EthereumNodeModule.kt
package com.github.wiljgriff.ethereumwallet.di.modules

import com.github.wiljgriff.ethereumwallet.ethereum.node.NodeBridge
import com.github.wiljgriff.ethereumwallet.ethereum.node.NodeDetails
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by williamgriffiths on 15/04/2017.
 */
@Module
class EthereumNodeModule {

    @Provides
    @Singleton
    fun provideNodeDetails(nodeBridge: NodeBridge): NodeDetails {
        return NodeDetails(nodeBridge)
    }

//    @Provides
//    @Singleton
//    fun provideNodeBridge() {
//        return
//    }
}